### ISSUE TITLE
Improve species (taxon) filtering in interactive search by adding "amigo_group_subset" subset

### ISSUE THREAD

**Issue by kltm** [geneontology/amigo#247]
The code for the improvement was finished in #209, now all that is needed is a subset with the name "model_slim", either as a separate ontology on in the taxon slim. Changed to over here to reflect that it is now a data issue that can be taken care of at individual sites rather than waiting on code. For GO, we'll aim at the next release.

**Comment by kltm** [comment]
@cmungall had this list suggested:


    Mammalia
    Vertebrata
    Fungi
    Metazoa
    Viridiplantae
    Eukaryota
    Archaea
    Bacteria

**Comment by cmungall** [comment]
Note the taxon field here still shows the full closure:

http://amigo2.berkeleybop.org/amigo/search/bioentity?q=*:*&fq=regulates_closure:%22GO:0022008%22&sfq=document_category:%22bioentity%22

which is impossible to browse to leaf taxa

It should be consistent with this:
http://amigo2.berkeleybop.org/amigo/term/GO:0022008

**Comment by kltm** [comment]
To clarify, it looks like a different issue--we switched one off of taxon_closure_label to just taxon_label. Open a new issue on switching all filters to taxon_label (easy config fix).

**Comment by kltm** [comment]
@cmungall temporary usability step has been separated as #250.

**Comment by cmungall** [comment]
Is there a blocker on me here? Should we add annotations to ncbitaxon to indicate what is in the slim?

Or maybe this should just be a yaml config list? The planteome folks will likely want more granularity in the plant taxa used.

**Comment by kltm** [comment]
As discussed, tagging @hdietze as well; apparently #209 got the owltools part done?
For the loader, we'll have two fields, names TBD: one for the full taxon closure and one for the slim.

**Comment by kltm** [comment]
Miracle of miracles, we already have this in the schema from a long time ago:
 "taxon_closure_subset"
 "taxon_closure_subset_label"
 "taxon_closure_subset_map"

**Comment by cmungall** [comment]
http://purl.obolibrary.org/obo/go/extensions/go-taxon-subsets.owl

Not yet imported into go-plus

**Comment by kltm** [comment]
404: Do you know how long these usually take to percolate out?

**Comment by cmungall** [comment]
every 30 mins. Next in at 2:40pm

**Comment by kltm** [comment]
Should we test this as a separate ontology in the load, or wait until it is imported?

**Comment by kltm** [comment]
Last question resolved by @hdietze .

**Comment by cmungall** [comment]
At some point we will want to make the ordering (at least partially) configurable rather than data-driven. This is too much for 2.4 but making a note here as the approach for the slim may overlap the approach for this.

**Comment by kltm** [comment]
Use "amigo_group_subet" for new option provided by @hdietze.

**Comment by kltm** [comment]
(Optional) flag will be, for example:
```
--solr-taxon-subset-name amigo_group_subset
```

**Comment by kltm** [comment]
Something is not right in our local tests--using the new code path seems to prevent the generation of *anything* taxon related, or at least getting to Solr (as a lot of the taxon info still goes by). To ensure that nothing unknown has change, removing 
```
--solr-taxon-subset-name amigo_group_subset
```
and
```
http://purl.obolibrary.org/obo/go/extensions/go-taxon-subsets.owl
```
and retrying.

**Comment by kltm** [comment]
Even after rolling back the two items above, I'm still not getting anything loaded besides taxon. amigo_group_subset no longer there, but the same 124 ncbi terms.

Well, this leads us back to either something subtle changed in owltools or something subtle at my end. I have been changing a large amount of metadata recently, but am a little lost on what that mechanism could be. A quicker test might just be going back a bit in owltools and seeing if there is an improvement.

**Comment by kltm** [comment]
Ever weirder. Re-trying with owlcollab/owltools@7f70c3b, I still get empties. I'm going to dig into metadata changes that I've made and see if there are any leads there.

**Comment by kltm** [comment]
@hdietze / @cmungall I've walked back to owlcollab/owltools@9c72e14 and still no luck, so maybe not owltools. I've also undone a bunch of stuff at the amigo end (although I'm not sure how any of that could prevent owltools from doing annotation loading--amigo stuff should only have an effect on schema generation and flexloader AFAICR), and no luck. I'll poke around some more in there.

I guess another avenue to pursue would be any changes to the ontologies? Anything gone on in there recently?

**Comment by cmungall** [comment]
Let's work on some junit unit/integration tests for this. I'm not following exactly what the issue is but seems it should be caught upstream.

**Comment by kltm** [comment]
More clear examination here in the morning with @hdietze. The immediate problems with losing the taxon and evidence closures have been traced back to ontology selection in the load, this has been cleared away. We're left with no subset closure yet, so we (he) will look toward the owltools end.

**Comment by hdietze** [comment]
correction the subset is named: `amigo_grouping_subset`
```
--solr-taxon-subset-name amigo_grouping_subset
```

**Comment by kltm** [comment]
Now just need to switch all displays over to the new one.
